clayui.com/src/components/IconSearch/index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * © 2020 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useMemo, useState} from 'react';
+
+const spritemap = '/images/icons/icons.svg';
+
+/**
+ * Component that allows searching through icons
+ * @param {string} label - Label of the search input field.
+ * @param {string} placeholder - Placeholder of the search input field.
+ * @param {Array[Object]} source - Path to the source JSON. Needs to follow pattern of [{name: String, aliases: Array<String>}]
+ * @param {string} type - Type of icons being provided, supports "icons" and "flags"
+ */
+const IconSearch = ({
+	label = 'Search Icons',
+	placeholder = 'Search Icons...',
+	source,
+	type = 'icons',
+}) => {
+	const [searchQuery, setSearchQuery] = useState('');
+	let list = [];
+
+	const filteredIcons = useMemo(() => {
+		const query = searchQuery.toLowerCase();
+
+		return source.filter(
+			({aliases, name}) =>
+				name.toLowerCase().includes(query) ||
+				aliases.some(alias => alias.toLowerCase().includes(query))
+		);
+	}, [searchQuery]);
```

Comment:
I think you would need to additionally add `source` to this array, since we would want this function to update if a new set of items was passed into the component.